### PR TITLE
Feat/domain detach zone

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-3729
| License          | BSD 3-Clause

## Description

Allow to detach DNS Zone 

### 🔖 TODO

- [x] TRANS-24839

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/zone/detach/detach.routing.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    resolve: {
+      goBack: /* @ngInject */ (goToZone) => goToZone,
+    },
+  });
+
+  $stateProvider.state('app.domain.alldom.zone.detach', {
+    url: '/detach',
+    component: 'domainZoneDetach',
+    resolve: {
+      goBack: /* @ngInject */ (goToZone) => goToZone,
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/domain/zone/detach/detach.routing.js:14]: same goBack in both states. may be put it in common resolvers?

File: packages/manager/apps/web/client/app/domain/zone/zone.routing.js [hunk 2]
```diff
@@ -1,6 +1,30 @@
+import get from 'lodash/get';
 import controller from './domain-zone-dns.controller';
 import template from './domain-zone-dns.html';
 
+const commonResolves = {
+  detachZoneOptions: /* @ngInject */ (
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/domain/zone/zone.routing.js:6]: why is this defined here if it is only used in the detach route. you can still define a common route in detach and use it.

Comment (marie-j) [reply packages/manager/apps/web/client/app/domain/zone/zone.routing.js:6]: @radireddy It is used here https://github.com/ovh/manager/pull/2833/files#diff-09710e5e28a23449aac1f9e2074ee072R32 to check if zone can be detached first

**Comment by antleblanc** [comment]
@marie-j Could you please resolve existing conflict? Thanks!

File: packages/manager/apps/web/client/app/domain/zone/detach/detach.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<oui-back-button on-click="$ctrl.goBack()">
+    <span data-translate="web_domain_zone_detach"></span>
+</oui-back-button>
+
+<div data-ovh-alert></div>
+
+<p data-translate="web_domain_zone_detach_description"></p>
+
+<ovh-manager-product-offers
+    on-error="$ctrl.onError(error)"
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/domain/zone/detach/detach.html:10]: data-

**Comment by marie-j** [comment]
> @marie-j Could you please resolve existing conflict? Thanks!

@antleblanc Conflicts solved :)

File: packages/manager/apps/web/client/app/domain/zone/detach/detach.html (hunk 3 above)
Comment (marie-j) [reply packages/manager/apps/web/client/app/domain/zone/detach/detach.html:10]: As it is a custom component, it is not mandatory

**Pull request merged**